### DISCUSSION TITLE
Option to enable album covers on discord presence

### DISCUSSION THREAD

**Discussion by chepgucciness** [pear-devs/pear-desktop#422] [Ideas]
Pretty new to all this, so I'm not sure how difficult this is, or if this has been asked already. But being able to see the album cover of the current song would be really cool!

**Comment by Araxeus** [comment]
There is a potential way to maybe do it if anyone has time and patience:
* Ask user to register a discord app and input it in config
* programmatically upload artwork to the specified appID
* refresh the presence with the uploaded artwork ID
* delete the artwork after song is done (if next song doesn't use the same artwork)

this plugin: https://github.com/maybeclean/musicbee-rich-presence does most of this and can be used for inspiration

also here is some doc about discord asset API: https://gist.github.com/SilverCory/99c99f2dbafb3cbdafe60edd9c9db121

notes:
> * Discord has a max size of 300 for assets which means they need to be deleted
> * Characters from other languages (Japanese for example) might not pass Discord's asset regex and fail to upload artwork)

**Comment by Araxeus** [comment]
This feature is available in https://github.com/th-ch/youtube-music/releases/tag/v1.16.0 🙂
